sc/social/like/browser/viewlets.py

```diff
@@ -74,6 +89,49 @@ def enabled(self):
             return True
         return self.helper.enabled(self.view)
 
+    @property
+    def via(self):
+        record = ISocialLikeSettings.__identifier__ + '.twitter_username'
+        try:
+            return api.portal.get_registry_record(record)
+        except InvalidParameterError:
+            return ''
+
+    @property
+    def canonical_url(self):
+        if ISocialMedia.providedBy(self.context):
+            return self.context.canonical_url
+        else:
+            # use current URL if the object don't provide the behavior
+            return self.url
+
+    def image_url(self):
+        """ Return url to image
+        """
+        img = self.image
+        if img:
+            return img.url
+        else:
+            return '{0}/logo.png'.format(self.site_url)
```

Comment:
that's an unexpected information, thank you for sharing.

anyway, do you plan to build a rocket? https://hackernoon.com/yes-python-is-slow-and-i-dont-care-13763980b5a1

:smile: